landscape/lib/apt/package/facade.py

```diff
@@ -269,15 +269,21 @@ def _get_internal_sources_list(self):
         sources_dir = apt_pkg.config.find_dir("Dir::Etc::sourceparts")
         return os.path.join(sources_dir, "_landscape-internal-facade.list")
 
-    def add_channel_apt_deb(self, url, codename, components=None):
+    def add_channel_apt_deb(self, url, codename, components=None,
+                            trusted=None):
         """Add a deb URL which points to a repository.
 
         @param url: The base URL of the repository.
         @param codename: The dist in the repository.
         @param components: The components to be included.
+        @param trusted: Whether validation shoule be skipped (if local).
         """
         sources_file_path = self._get_internal_sources_list()
-        sources_line = "deb %s %s" % (url, codename)
+        source_options = ""
+        if trusted is not None:
```

Comment:
We probably should also confirm that url.startswith('file:')  before trying to add trusted=yes right?   Even though it looks like all the calls are called with file://%s

Reply:
Although the only callsite is add_channel_deb_dir() which is a file:// URL, I think it would be a safe check to do as that flag could be easily misused. Consider it done.